### ISSUE TITLE
[Bug] Setting --project-dir also changes the working directory

### ISSUE THREAD

**Issue by aBBDnGus** [dbt-labs/dbt-core#7445]
### Is this a new bug in dbt-core?

- [X] I believe this is a new bug in dbt-core
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

When executing the command `dbt run --project-dir mydir`, this also changes the working directory. I have the following directory structure:
```
project
| - models
| - mydir
    | - dbt_projects.yml
```
When I execute the command, the models are searched in `mydir` instead of `project`. There is no way to explicitly set the working directory.

Additional error in the documenation https://docs.getdbt.com/reference/dbt_project.yml
> By default, dbt will look for dbt_project.yml in your current working directory and its parents, but you can set a different directory using the --profiles-dir flag.

`--profiles-dir` should be replaced by  `--project-dir`

### Expected Behavior

When executing the command `dbt run --project-dir mydir`, I expect that the file `mydir/dbt_project.yml` is used, but the working directory is still the directory, in which I execute the command.

### Steps To Reproduce

1. Execute `dbt run --project-dir mydir`

### Environment

```markdown
- OS: Windows 10
- Python: 3.9.12
- dbt: 1.4.5
```

**Comment by dbeatty10** [comment]
Good catch on the docs @aBBDnGus ! Opened a PR to fix it here: https://github.com/dbt-labs/docs.getdbt.com/pull/3256

### Default project folder structure

> When I execute the command, the models are searched in `mydir` instead of `project`. 

It's looking in `my_dir` because the default directory structure that dbt expects looks like this:
```
.
└── my_dir
    ├── analyses
    ├── dbt_project.yml
    ├── macros
    ├── models
    ├── packages.yml
    ├── seeds
    ├── snapshots
    └── tests
```

rather than what you described:
```
.
└── project
    ├── models
    └── mydir
        └── dbt_project.yml
```

### Overriding the default folder names and locations

> There is no way to explicitly set the working directory.

You _can_ actually override the default [`model-paths`](https://docs.getdbt.com/reference/dbt_project.yml) in your `dbt_project.yml` file!

Although your example is not a _typical_ directory structure, it should still work -- as long as you configure [`model-paths`](https://docs.getdbt.com/reference/dbt_project.yml) in your `dbt_project.yml` file (and any other `*-paths` that differ from the defaults).

`project/mydir/dbt_project.yml`
```
name: "my_dbt_project"
version: "1.0.0"
config-version: 2
profile: "my_profile_name"  # 👈 you'll need to update this to your profile name!

model-paths: ["../models"]
```

`project/models/sample_model.sql`
```
select 0 as id
```

Here's the command to run it:
```shell
dbt run --project-dir project/mydir 
```

Could you try setting [`model-paths`](https://docs.getdbt.com/reference/dbt_project.yml) and see how it works for you? Or try out the default directory structure (which may be a bit easier for you to work with)?

**Comment by aBBDnGus** [comment]
Thanks for the quick response, @dbeatty10!
The version with changing the `model-paths` would also work. But I thought that my suggestion is more elegant.

Background for my request: We have two databases, where one is a replica of the other, i.e. both databases have the same tables. In the `dbt_project.yml`, we specify the permissions for the views. As both databases have different users, the `dbt_project.yml` is different, but the models are the same. Therefore we want to have one directory with all the models and two different projects which handle these models.

**Comment by dbeatty10** [comment]
Thanks for the background @aBBDnGus -- that helps me give you two other suggestions for how you can approach this:

1. Use separate [targets](https://docs.getdbt.com/docs/core/connection-profiles#understanding-targets-in-profiles) for each of your two databases; or
2. Create a ([local](https://docs.getdbt.com/docs/build/packages#local-packages)) [dbt package](https://docs.getdbt.com/guides/legacy/building-packages) for your common models 

### Recommendation
This sounds like a job for **separate targets**.

I'm guessing that # 1 is the way that you'll want to go because it makes sense when you want two different databases to be the "same" with only slight differences (e.g., dev vs. production environments).

Option # 2 is best when there is a small subset of models, macros, etc that you want to share across multiple downstream dbt projects (but each downstream project still has their own unique models, macros, etc).

### Example using separate targets

Let's assuming you have the default directory structure like this:
```
.
└── my_project
    ├── dbt_project.yml
    ├── macros
    └── models
```

Then creating two replicas is as simple as:
1. Creating multiple targets in your `profiles.yml`
2. Specifying your desired target using the `--target` CLI flag

#### 1) Creating two different targets in your `profiles.yml` file (one for each replica):

`profiles.yml`
```yaml
my_profile_name:
  target: replica_1  # 👈 this is the default target that dbt uses when `--target` is not specified
  outputs:
    replica_1:
      type: postgres
      dbname: 'my_database_name'
      schema: 'replica_1'
      # ...
    replica_2:
      type: postgres
      dbname: 'my_database_name'
      schema: 'replica_2'
      # ...
```

If you want the databases to be different rather than your schemas, you'd just make that change within the `profiles.yml`.

#### 2) To build each of your different replicas, you'd do the following:
```shell
dbt build --target replica_1
dbt build --target replica_2
```

Since `replica_1` was configured as the default target within `profiles.yml`, you could also do this instead:
```shell
dbt build
dbt build --target replica_2
```

Could you give that a shot and let me know how that goes?

**Comment by aBBDnGus** [comment]
Thanks! Unfortunately, two targets do not work. The `dbt_project.yml` contains the following snippet:
```
+grants:
  select: user_replica_1
```
`user_replica_1` exists only in `replica_1`, `user_replica_2` exists only in `replica_2`. Executing this on both databases gives an error, because the user does not exist. We also cannot synchronize the users between the databases.
But I think, the options "Overriding the default folder names" and "local dbt package" should work.
